src/lib/io/sol-memmap-storage-impl-zephyr.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <zephyr.h>
+#include <flash.h>
+#include <device.h>
+
+#include "sol-memmap-storage-impl.h"
+
+#include "sol-buffer.h"
+#include "sol-mainloop.h"
+#include "sol-str-slice.h"
+#include "sol-str-table.h"
+#include "sol-util-internal.h"
+#include "sol-vector.h"
+
+static struct device *flash_dev = NULL;
```

Comment:
Thinking better, for the Arduino 101 SoC, which I target here, there's no sense on having more than one attached (it won't change). If we're having broader support later, we could fix this accordingly.